From 81f3b748655dc4c57d75e4a49574ba9dca9a93d7 Mon Sep 17 00:00:00 2001
From: Amarnath Valluri <amarnath.valluri@intel.com>
Date: Tue, 28 Jul 2015 15:32:11 +0300
Subject: [PATCH] pam_timestamp: Avoid leaking file descriptor.

  * modules/pam_timestamp/hmacsha1.c(hmac_key_create):
    close 'keyfd' when failed to own it.

  Upstream-Status: Pending

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>
---
 modules/pam_timestamp/hmacsha1.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/pam_timestamp/hmacsha1.c b/modules/pam_timestamp/hmacsha1.c
index 573ecf3..3f41106 100644
--- a/modules/pam_timestamp/hmacsha1.c
+++ b/modules/pam_timestamp/hmacsha1.c
@@ -73,6 +73,7 @@ hmac_key_create(pam_handle_t *pamh, const char *filename, size_t key_size,
 
 	 if (fchown(keyfd, owner, group) == -1) {
 		pam_syslog(pamh, LOG_ERR, "Cannot chown %s: %m", filename);
+		close(keyfd);
 		return;
 	}
 
-- 
1.9.1

